pandas/tseries/index.py

```diff
@@ -756,7 +756,7 @@ def to_datetime(self, dayfirst=False):
     def astype(self, dtype):
         dtype = np.dtype(dtype)
 
-        if dtype == np.object_:
+        if dtype == np.object_ or dtype == np.str:
             return self.asobject
         elif dtype == _INT64_DTYPE:
             return self.asi8.copy()
```

Comment:
I suspect this might be the same issue with ``TimedeltaIndex`` and ``PeriodIndex``, pls add a fix and tests as well (in ``tseries/tests/test_base``